pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ProducerMultiSchemaMode.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api;
+
+/**
+ * Multiple schema mode for producer,
+ * indicates to enable, disable, or automatically enable multiple schema.
+ */
+public enum ProducerMultiSchemaMode {
+    /**
+     * Automatically enable multiple schema for producer if needed.
+     */
+    Auto,
+
+    /**
+     * Enable multiple schema mode for producer, with this mode producer can send a message
+     * with different schema from that specified just when it is created.
+     */
+    Enabled,
```

Comment:
What is the difference between Auto and Enabled? Auto seems that nothing is different from Enabled

Reply:
Enabled mode would lead to more check in some path. Auto means to discard those checks and only enable it when need.

Reply:
I see you use ProducerMultiSchemaMode.AUTO to avoid more check if schema of message is equals schema of producer. But i think this is an internal logic. Do we need to be exposed to users?

I think users just need to producer.enableMultiSchema(true or false) and we can have 3 internal flags(Auto Enabled Disabled). 

@sijie @jiazhai Please also help me to see if I understand it correctly.

Reply:
It's great to separate these two parts. I'll fix it. Thanks.

Reply:
@codelipenghui it's fixed, PTAL.